src/Http/Handler/ResponseHandlerError.php

```diff
@@ -45,37 +29,47 @@ class ResponseHandlerError extends ResponseHandlerBase
      */
     public function execute(ResponseInterface $response): ResponseInterface
     {
-        $contentType = $response->getHeaderLine(self::HEADER_CONTENT_TYPE);
+        if ($response->getStatusCode() !== self::STATUS_CODE_OK){
+            throw ExceptionFactory::createExceptionForResponse(
+                $this->fetchErrorMessages($response),
+                $response->getStatusCode()
+            );
+        }
+
+        return $response;
+    }
 
-        if ($response->getStatusCode() === self::STATUS_CODE_OK) {
-            return $response;
-        } else {
-            if ($contentType === self::HEADER_CONTENT_TYPE_APPLICATION_JSON) {
-                $responseBody = $response->getBody();
-                $responseJson = \GuzzleHttp\json_decode($responseBody, true);
+    /**
+     * @param ResponseInterface $response
+     *
+     * @return string[]
+     */
+    private function fetchErrorMessages(ResponseInterface $response): array
+    {
+        $responseBody = $response->getBody();
+        $responseBodyInJson = json_decode($responseBody, true);
 
-                $errorDescriptions = [];
+        if ($responseBodyInJson != false){
```

Comment:
@OGKevin `!==`. lets not fall in the php-weak-typing-trap

Reply:
Ahh 🙈 nice spot !